### DISCUSSION TITLE
Is there a way to have the tooltip "vertical crosshair" and "snap tooltip to datum x" to be available on a line chart, NOT <XYChart />

### DISCUSSION THREAD

**Discussion by hyfydistro** [airbnb/visx#1300] [General]
I want to have the same "snap" effect too but not in the prebuilt "XYChart".

I am making a multiline chart and I would like them all in sync aswell. How do I address the snapping issue? When I cross a point of the datum, I have a circle shape that just floats until I'm on the next nearest datum point.

**Comment by williaster** [comment]
hey @hyfydistro 👋  thanks for checking out visx. I'm going to move this to discussions since it's a usage question, not an issue.

To accomplish the "snap to nearest point" behavior, you need to 
- find the nearest data point to the pointer event
- apply the xScale to that data point's x-value to get the location of the point on the chart
- use that scaled x-value to position the tooltip (rather than the x-coordinate of the pointer event itself)


Here's a [modified `/areas` demo](https://codesandbox.io/s/charming-fog-yuexs?file=/Example.tsx) which shows snapping behavior 
<img src="https://user-images.githubusercontent.com/4496521/127903844-2820a48d-1abf-4a65-890d-e3231e70848f.gif" width="500" />

For comparison here is the [existing demo](https://airbnb.io/visx/areas) which uses the pointer event for the x-coordinate
<img src="https://user-images.githubusercontent.com/4496521/127903882-ab0a3006-b18d-4c55-888c-37c48ccc2c23.gif" width="500" />
